src/escapedHashtags.js

```diff
@@ -25,90 +25,16 @@
  * TODO: only use escaped hashtag form internally for invalid xpaths
  */
 define([
+    'underscore',
     'vellum/xpath',
 ], function(
+    _,
     xpath
 ) {
     var OUTSIDE_HASHTAG = 0,
         INSIDE_HASHTAG = 1,
         DELIMITER = "`",
-        defaultParser = xpath.createParser(xpath.makeXPathModels({}, {}));
-
-    function toXPath(input, xpathParser) {
-        xpathParser = xpathParser || defaultParser;
-        return transform(input, function(input) {
-            return xpathParser.parse(input).toXPath();
-        });
-    }
-
-    function toHashtag(input, xpathParser) {
-        xpathParser = xpathParser || defaultParser;
-        return transform(input, function(input) {
-            return xpathParser.parse(input).toHashtag();
-        });
-    }
-
-    /*
-     * hopefully robust parser that transforms any input (whether hashtag,
-     * xpath, or escaped hashtag) into escaped hashtag used by internal structures
-     */
-    function toEscapedHashtag(input, xpathParser) {
-        if (!input) { return input; }
-        xpathParser = xpathParser || defaultParser;
-
-        var hashtag = transform(input);
-
-        try {
-            var parsed = xpathParser.parse(hashtag);
-            parsed = xpathParser.parse(parsed.toHashtag());
-            return transformHashtags(parsed, xpathParser.models, function(input) {
```

Comment:
parse/render/parse/render simplified to a single parse/render: https://github.com/dimagi/Vellum/pull/743/files#diff-252bc0c8b00cd8556f4f3c24d84cd59fR154